Server/mods/deathmatch/logic/CMainConfig.cpp

```diff
@@ -133,6 +133,29 @@ bool CMainConfig::Load()
         return false;
     }
 
+    // Grab rules
+    CXMLNode* pNode = nullptr;
+    std::size_t uiCurrentIndex = 0;
+    do
+    {
+        // Grab the current script node
+        pNode = m_pRootNode->FindSubNode("rule", uiCurrentIndex++);
+        if (pNode)
+        {
+            // Grab its "name" attribute
+            CXMLAttribute* pAttribute = pNode->GetAttributes().Find("name");
+            SString        strName = pAttribute ? pAttribute->GetValue() : SString{};
+
+            // Grab its "value" attribute
+            pAttribute = pNode->GetAttributes().Find("value");
+            SString strValue = pAttribute ? pAttribute->GetValue() : SString{};
+
+            // Ignore if name or value are empty
+            if (!strName.empty() && !strValue.empty())
+                m_RulesForASEMap[std::move(strName)] = std::move(strValue);
+        }
+    } while (pNode);
```

Comment:
`while` would be better here:
```cpp
while(pNode = m_pRootNode->FindSubNode("rule", uiCurrentIndex++))
{
    // Grab its "name" attribute
    CXMLAttribute* attribute = pNode->GetAttributes().Find("name");
    SString        strName = attribute ? attribute->GetValue() : SString{};

    // Grab its "value" attribute
    attribute = pNode->GetAttributes().Find("value");
    SString strValue = attribute ? attribute->GetValue() : SString{};

    // Ignore if name or value are empty
    if (!strName.empty() && !strValue.empty())
        m_RulesForASEMap[std::move(strName)] = std::move(strValue);
}
```

Reply:
I agree, you should use `while` instead of `do ... while`, unless you need the body of the loop to execute at least once before the condition is evaluated.

Reply:
Ok!